tests/v2/validation/workloads/workload_test.go

```diff
@@ -0,0 +1,67 @@
+package workloads
+
+import (
+	"testing"
+
+	projectsapi "github.com/rancher/rancher/tests/v2/actions/projects"
+	deployment "github.com/rancher/rancher/tests/v2/actions/workloads/deployment"
+	"github.com/rancher/shepherd/clients/rancher"
+	management "github.com/rancher/shepherd/clients/rancher/generated/management/v3"
+	"github.com/rancher/shepherd/extensions/charts"
+	"github.com/rancher/shepherd/extensions/clusters"
+	"github.com/rancher/shepherd/pkg/session"
+	log "github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type WorkloadTestSuite struct {
+	suite.Suite
+	client  *rancher.Client
+	session *session.Session
+	cluster *management.Cluster
+}
+
+func (connection *WorkloadTestSuite) TearDownSuite() {
+	connection.session.Cleanup()
+}
+
+func (connection *WorkloadTestSuite) SetupSuite() {
+	connection.session = session.NewSession()
+
+	client, err := rancher.NewClient("", connection.session)
+	require.NoError(connection.T(), err)
+
+	connection.client = client
+
+	log.Info("Getting cluster name from the config file and append cluster details in connection")
+	clusterName := client.RancherConfig.ClusterName
+	require.NotEmptyf(connection.T(), clusterName, "Cluster name to install should be set")
+
+	clusterID, err := clusters.GetClusterIDByName(connection.client, clusterName)
+	require.NoError(connection.T(), err, "Error getting cluster ID")
+
+	connection.cluster, err = connection.client.Management.Cluster.ByID(clusterID)
+	require.NoError(connection.T(), err)
+}
+
+func (c *WorkloadTestSuite) TestWorkloadDeployment() {
+	subSession := c.session.NewSession()
+	defer subSession.Cleanup()
+
+	_, namespace, err := projectsapi.CreateProjectAndNamespace(c.client, c.cluster.ID)
+	require.NoError(c.T(), err)
+
+	ngInxDeployment, err := deployment.CreateDeployment(c.client, c.cluster.ID, namespace.Name, 1, "", "", false, false)
+	require.NoError(c.T(), err)
+
+	err = charts.WatchAndWaitDeployments(c.client, c.cluster.ID, namespace.Name, metav1.ListOptions{
```

Comment:
`WatchAndWaitDeployments` is taken care in deployment as well. so we dont have to add his.